dvc/command/diff.py

```diff
@@ -111,14 +111,15 @@ def _show_diff(diff, hide_missing=False):
         if not sum(summary.values()):
             return None
 
-        fmt = (
-            "files summary: {added} added, {deleted} deleted,"
-            " {renamed} renamed, {modified} modified"
-        )
-        if not hide_missing:
-            fmt += ", {not in cache} not in cache"
```

Comment:
After these changes we won't be accounting for `hide_missing`. The `not in cache` state should be hidden if the value is `0` or `hide_missing` is true

Reply:
It will, since `not in cache` is added to the list of states.

Reply:
The issue is that `not in cache` is always in the list of states.

If `hide_missing` is True, that state should be hidden from the CLI output. But with your change, the `not in cache` state will always be appended to `states_values` (and output to the CLI) as long as the value for it is not zero.

`not in cache` state needs to be handled slightly differently than the other states - it should only be appended to `state_values` if the value is not zero AND `hide_missing` is False.

Reply:
Look at line 83 of this file - `not in cache` is only added to the `states` list if the flag is set to `False`.

Reply:
ah I see now, I missed that and was thinking states came directly from the diff